### PR TITLE
Keep comments when removing unnecessary lambda expressions

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/71300

### CODE REVIEW

**Comment by dibarbet** [comment]
> Fixes #71300

Is 71300 the right issue?  Seems already closed and not related

File: src/Analyzers/CSharp/CodeFixes/RemoveUnnecessaryLambdaExpression/CSharpRemoveUnnecessaryLambdaExpressionCodeFixProvider.cs [hunk 1]
```diff
@@ -52,8 +58,32 @@ protected override Task FixAllAsync(
...
     }
+
+    private static IEnumerable<SyntaxTrivia> TakeComments(SyntaxTriviaList triviaList)
+    {
+        var lastComment = triviaList.Last(t => t.IsSingleOrMultiLineComment());
+        var lastIndex = triviaList.IndexOf(lastComment) + 1;
+        if (lastIndex < triviaList.Count && triviaList[lastIndex].IsEndOfLine())
+            lastIndex++;
+
+        return triviaList.Take(lastIndex);
```
Comment (dibarbet) [inline src/Analyzers/CSharp/CodeFixes/RemoveUnnecessaryLambdaExpression/CSharpRemoveUnnecessaryLambdaExpressionCodeFixProvider.cs:87]: didn't quite get why we need this filtering at a glance - mind adding a comment on it?

**Review by dibarbet — APPROVED** [review]

File: src/Analyzers/CSharp/CodeFixes/RemoveUnnecessaryLambdaExpression/CSharpRemoveUnnecessaryLambdaExpressionCodeFixProvider.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/CodeFixes/RemoveUnnecessaryLambdaExpression/CSharpRemoveUnnecessaryLambdaExpressionCodeFixProvider.cs:87]: the idea is that we want to grab up through the last comment and the newline that follows and only move those over if it exists.  we don't want to generally move things like when the user only has blank lines above the inner call.  i will doc :)

**Comment by drewnoakes** [comment]
> > Fixes #71300
> 
> Is 71300 the right issue? Seems already closed and not related

Should be #60741

**Pull request merged**